packages/core/core-flows/src/definition/cart/steps/validate-cart-payments.ts

```diff
@@ -0,0 +1,45 @@
+import { CartWorkflowDTO } from "@medusajs/types"
+import { isPresent, MedusaError, PaymentSessionStatus } from "@medusajs/utils"
+import { createStep, StepResponse } from "@medusajs/workflows-sdk"
+
+interface StepInput {
+  cart: CartWorkflowDTO
+}
+
+export const validateCartPaymentsStepId = "validate-cart-payments"
+export const validateCartPaymentsStep = createStep(
+  validateCartPaymentsStepId,
+  async (data: StepInput) => {
+    const {
+      cart: { payment_collection: paymentCollection },
+    } = data
+
+    if (!isPresent(paymentCollection)) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `Payment collection has not been initiated for cart`
+      )
+    }
+
+    // We check if any of these payment sessions are present in the cart
+    // If not, we throw an error for the consumer to provide a processable payment session
+    const processablePaymentStatuses = [
+      PaymentSessionStatus.PENDING,
+      PaymentSessionStatus.REQUIRES_MORE,
+    ]
+
+    const paymentsToProcess = paymentCollection.payment_sessions?.filter((ps) =>
+      processablePaymentStatuses.includes(ps.status)
+    )
+
+    if (!paymentsToProcess?.length) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `Payment sessions are required to complete cart`
+      )
+    }
+
+    // TODO: Should we be able to authorize multiple payment sessions for a cart?
```

Comment:
Don't think we should, but since this is more a generic validation step, maybe this should return all the payments to process, and then in our specific workflow, we select `[0]`, because we know there will only ever be one (for the time being).